src/formatting/newline_style.rs

```diff
@@ -1,5 +1,10 @@
 use crate::NewlineStyle;
 
+const LINE_FEED: char = '\n';
+const CARRIAGE_RETURN: char = '\r';
+const WINDOWS_NEWLINE: &str = "\r\n";
+const UNIX_NEWLINE: &str = "\n";
```

Comment:
Should this not be `&'static str`?

Reply:
Not sure if these comments are for me, as the PR author, or for the potential merge.  In any case, if I understand correctly, comparing to C, rust `const` is somewhat similar to C `#define` while rust `static` is similar to C`global` or `static` variable.  If this is true, isn't more effective in this case to use `const`?

Reply:
Apologies if it wasn't clear, but yes the comments were for you as the author. I think the annotation here is probably fine. I was mostly wondering why we couldn't return `&'static str` from `get_newline_string` and `get_newline_string_of_text`, and was wondering if maybe that had something to do with the `&str` annotation here forcing you to return a `String` from those functions.

Reply:
Understood. I don't think we need to make any changes to the definitions here, unless they somehow prevent us from returning `&str` or `&'static str` from the new `get_newline_string` or `get_newline_string_of_text` that you added in this PR. See my comments bellow 👇🏼

Reply:
OK. Changed.
Somehow I missed your point below about changing only the return type, not the `const` definitions.